### ISSUE TITLE
New fasttext functionality (print-sentence-vector and quantize)

### ISSUE THREAD

**Issue by minimaxir** [salestock/fastText.py#122]
[fasttext](https://github.com/facebookresearch/fastText) announced new functionality today, including a `print-sentence-vector` function and a `quantize` function, both of which would be very useful to the Python port.

**Comment by spate141** [comment]
Also loading new quantized models of `.ftz` format! Which would save a great amount of memory in loading large models.
Thanks :)

**Comment by pyk** [comment]
@minimaxir @spate141 Yup yup. I will try to work on this over the next weekend. PR also welcome! 😄

**Comment by ErnstTmp** [comment]
@pyk do you still need help on the topic of loading new FT-bin files?

**Comment by tarrekshaban** [comment]
"Edit: Also we can't load newer .bin models created with updated fastText library using fasttext.load_model(model.bin)"

How should we load the model than if we created it with the updated fastText library?

**Comment by pyk** [comment]
@ErnstTmp Yup, feel free to send PR. I can't manage to allocate my time to ship the update yet. Currently I'm updating the [C++ wrapper](https://github.com/salestock/fastText.py/blob/master/fasttext/interface.cc) to be able to work with the latest `fasttext(1)`.

@tarrekshaban Sorry, currently this package is not able to load the model that generated by the latest version of fastText

**Comment by spate141** [comment]
@pyk Any progress in new features update? Happy to help if needed anywhere! :)

**Comment by namp** [comment]
Yes, I was wondering if there's any progress on this issue since it renders the API unusable with the latest fasttext models.

**Comment by kskrishnasangeeth** [comment]
I was also having the same issue. Looks like fix is needed for https://github.com/salestock/fastText.py/blob/2e21e38ca1ebcbb658e884c68206ed6f2182a666/fasttext/interface.cc#L286

**Comment by pyk** [comment]
Sorry guys for late reply. 

@kskrishnasangeeth Yup! currently I'm working on the [`interface.cc`](https://github.com/salestock/fastText.py/blob/master/fasttext/interface.cc) to be able to work with the latest version of fastText. I will push the changes tonight :+1:

**Comment by pyk** [comment]
Btw you can see the progress here #128

**Comment by tiru1930** [comment]
Is this issue got fixed?. if it is do i need to update my  fasttext pip ,to support??

**Comment by namp** [comment]
Any updates on this?

**Comment by aclevine** [comment]
@namp 
basic interface and model loading has been setup for the new version of fasttext at this point.  there's an issue with the qmatrix class facebook added that was causing a error in the wrapper when you deallocated a model from memory that was causing some road blocks.  Basics are still there if you want to go to the update branch and do a bit of coding, was able to pretty easily add back in prediction functionality myself. But, you'll probably have to wait a bit for pyk to add back in everything.

**Comment by spate141** [comment]
@aclevine Can you specify what exact changes needs to make in terms of loading trained models with new fastText library? Is that only [this one..?](https://github.com/salestock/fastText.py/pull/128#issuecomment-308458212)

**Comment by namp** [comment]
I'm sorry but I don't see the file fasttext/cpp/src/qmatrix.cc mentioned in the [#128 ](https://github.com/salestock/fastText.py/pull/128)fix so that I can make[ the suggested change](https://github.com/salestock/fastText.py/pull/128#issuecomment-308458212)

**Comment by wangyizhen** [comment]
**Firstly,** just clone from the github. and complie the fasttext (C++ version) from the cpp folders that have been cloned

git clone https://github.com/salestock/fastText.py.git
cd fastText
python setup.py install

**then,** you will installed fasttext python package success.

**Finally** , just to compile the fasttext c++ source code.

cd fasttext/cpp
make
you will compile success.
